assets/javascripts/kitten/karl/menus/karl-menu.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import classNames from 'classnames'
+import { KarlMenuHeader } from 'kitten/karl/headers/karl-header'
+
+export class KarlMenu extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { opened: false }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick() {
+    this.setState({ opened: !this.state.opened })
+  }
+
+  renderOpacity() {
+    if (this.state.opened)
+      return <div className="karl-Opacity" onClick={ this.handleClick } />
+  }
+
+  render() {
+    const karlMenuHeaderClassName = classNames(
+      'karl-Menu',
+      {
+        'is-opened': this.state.opened,
+      },
+    )
+
+    return (
+      <div>
+        <div className={ karlMenuHeaderClassName }>
+          <KarlMenuHeader onButtonClick={ this.handleClick } />
+          <KarlMenuContent
+            onMenuItemClick={ this.handleClick }
+            brandingCategories={ this.props.brandingCategories }
+            componentCategories={ this.props.componentCategories } />
+        </div>
+
+        { this.renderOpacity() }
+      </div>
+    )
+  }
+}
+
+export class KarlMenuContent extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.renderBrandCategory = this.renderBrandCategory.bind(this)
+    this.renderComponentCategory = this.renderComponentCategory.bind(this)
+  }
+
+  renderBrandCategory(category) {
+    return <KarlMenuCategory key={ category.id }
+                             href="branding"
+                             onMenuItemClick={ this.props.onMenuItemClick }
+                             { ...category } />
+  }
+
+  renderComponentCategory(category) {
+    return <KarlMenuCategory key={ category.id }
+                             href="components"
+                             onMenuItemClick={ this.props.onMenuItemClick }
+                             { ...category } />
+  }
+
+  render() {
+    return (
+      <nav className="karl-Menu__content">
+        <div className="k-u-clearfix"></div>
+        <a href="branding" className="karl-Menu__listTitle">
+          Branding
+        </a>
+        <ul>
+          { this.props.brandingCategories.map(this.renderBrandCategory) }
+        </ul>
+        <a href="components" className="karl-Menu__listTitle">
+          Components
+        </a>
+        <ul>
+          { this.props.componentCategories.map(this.renderComponentCategory) }
+        </ul>
+      </nav>
+    )
+  }
+}
+
+KarlMenuContent.defaultProps = {
+  brandingCategories: [],
+  componentCategories: [],
+}
+
+class KarlMenuCategory extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { opened: false }
+
+    this.renderComponent = this.renderComponent.bind(this)
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  renderComponent(component) {
+    return (
+      <li key={ component.path }>
+        <a href={ `${this.props.href}#${component.path}` }
```

Comment:
Ah bon ? Il me semblait pas qu'on faisait ce genre d'espaces dans les interpolations. C.f. "No spacing in template strings" de https://standardjs.com/rules.html